Source/PINMemoryCache.m

```diff
@@ -193,6 +206,27 @@ - (void)trimMemoryToDate:(NSDate *)trimDate
     }
 }
 
+- (void)removeExpiredObjects
+{
+    [self lock];
+        NSDictionary<NSString *, NSDate *> *dates = [_dates copy];
+        NSDictionary<NSString *, NSNumber *> *ageLimits = [_ageLimits copy];
+    [self unlock];
+
+    NSDate *now = [NSDate date];
+    for (NSString *key in ageLimits) {
+        NSDate *accessDate = dates[key];
+        NSTimeInterval ageLimit = [_ageLimits[key] doubleValue] ?: self->_ageLimit;
```

Comment:
Accessing these ivars is unsafe, gather `self->_ageLimit` as `globalAgeLimit` in the lock above and use `ageLimits` here?

Reply:
Done.